ext/opentelemetry-ext-docker-tests/tests/postgres/test_psycopg_functional.py

```
@@ -0,0 +1,130 @@
+        spans = self._span_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 2)
+        for span in spans:
+            if span.name == "rootSpan":
+                root_span = span
+            else:
+                child_span = span
+            self.assertIsInstance(span.start_time, int)
+            self.assertIsInstance(span.end_time, int)
+        self.assertIsNot(root_span, None)
+        self.assertIsNot(child_span, None)
+        self.assertEqual(root_span.name, "rootSpan")
+        self.assertEqual(child_span.name, "postgresql.opentelemetry-tests")
+        self.assertIsNotNone(child_span.parent)
+        self.assertEqual(child_span.parent.name, root_span.name)
+        self.assertIs(child_span.kind, trace_api.SpanKind.CLIENT)
+        self.assertEqual(
+            child_span.attributes["db.instance"], POSTGRES_DB_NAME
+        )
+        self.assertEqual(child_span.attributes["net.peer.name"], POSTGRES_HOST)
+        self.assertEqual(child_span.attributes["net.peer.port"], POSTGRES_PORT)
+
+    def test_execute(self):
+        """Should create a child span for execute method
+        """
+        with self._tracer.start_as_current_span("rootSpan"):
+            self._cursor.execute(
+                "CREATE TABLE IF NOT EXISTS test (id integer)"
+            )
+            self._connection.commit()
+        self.validate_spans()
+
+    def test_executemany(self):
+        """Should create a child span for executemany
+        """
+        with self._tracer.start_as_current_span("rootSpan"):
+            data = ("1", "2", "3")
+            stmt = "INSERT INTO test (id) VALUES (%s)"
+            self._cursor.executemany(stmt, data)
+        self.validate_spans()
+
+    def test_callproc(self):
+        """Should create a child span for callproc
+        """
+        with self._tracer.start_as_current_span("rootSpan"):
+            try:
+                self._cursor.callproc("test", ())
+            except Exception as err:
```

does this exception need to be caught? seems like it's fine to let it raise up.

Even if it's unrelated to our code, exceptions are a separate code path that we should exercise explicitly.

---

Stored procedure doesn't exist so callproc will fail but Span will be created, I want to avoid as much pre setup as possible for this tests

---

You might want an `assertRaises` here instead then, the tests generally shouldn't print anything.